fiftyone/core/annotation.py

```diff
@@ -78,3 +82,119 @@ class AnnotationResults(BaseRunResults):
     """Base class for annotation run results."""
 
     pass
+
+
+def compute_annotation_schema(collection, field_name):
+    """Compute the annotation schema for a collection's field
+
+    An annotation schema is defined by a type. A field type and an annotation
+    type informs the annotation form type and allowed values
+
+    Annotation types are:
+        - checkbox
+        - input
+        - select
+        - radio
+        - text
+        - tags
+
+    Args:
+        collection: a :class:`SampleCollection`
+        field_name: the field name to process
+
+    Raises:
+        ValueError: if the field does not exists or annotation for its
+        field type is not supported
+
+    Returns:
+        an annotation schema dictionary
+
+    """
+    if collection.media_type != fom.IMAGE:
+        raise ValueError("only image datasets are supported")
+
+    if field_name is None:
+        raise ValueError("field_name is required")
+
+    field = collection.get_field(field_name)
+    if field is None:
+        raise ValueError(f"field '{field_name}' does not exist")
+    is_list = isinstance(field, fof.ListField)
+    if is_list:
+        field = field.field
+
+    if isinstance(field, fof.StringField):
+        try:
+            return {
+                "default": None,
+                "type": "checkbox" if is_list else "select",
+                "values": collection.distinct(field_name),
+            }
+        except:
+            # too many distinct values
+            return {"default": None, "type": "input"}
+
+    if is_list:
+        raise ValueError(f"unsupported annotation field {field}")
+
+    if isinstance(field, fof.BooleanField):
+        return {
+            "default": None,
+            "type": "radio",
+            "values": [True, False, None],
+        }
+
+    if isinstance(
+        field,
+        (
+            fof.DateField,
+            fof.DateTimeField,
+            fof.ObjectIdField,
+            fof.FloatField,
+            fof.IntField,
+        ),
+    ):
+        return {"type": "input", "default": None}
+
+    if not isinstance(field, fof.EmbeddedDocumentField):
+        raise ValueError(f"unsupported annotation field {field}")
+
+    if not field.document_type in (
```

Comment:
Do we need to raise an error in this method? Or can we go head and support all label types in this method and only ignore unsupported fields in the HA UX?